File: permutation.java
public class permutation {
    public void main(String[] args){
        System.out.printf("%s\n",permutation("abcde", "edcba"));
        System.out.printf("%s\n", permutation("helloworld", "helloporld"));
    }
    boolean permutation(String s, String t) {
        if (s.length() != t.length()){
            return false;
        }

        int[] letter_array = new int[128];

        char[] s_array = s.toCharArray();
        for (char c : s_array) {
            letter_array[c]++;
        }

        for (int i = 0; i < t.length(); i++) {
            int c = t.charAt(i);
            letter_array[c]--;
            if (letter_array[c] < 0){
                return false;
            }
        }

        return true;
    }
}

File: permutation.py
# given two strings, write a method to decide if one is permutation of the other

def check_permutation(string1, string2):
    if len(string1) != len(string2):
        return False

    return sorted(string1) == sorted(string2)


string1 = 'abcde'
string2 = 'edcba'
string3 = 'alkdakjfe'
string4 = 'alaleeee4'
print(check_permutation(string1, string2))
print(check_permutation(string3, string4))

